src/pokemon_storage_system.c

```diff
@@ -6438,10 +6438,23 @@ static void SetPlacedMonData(u8 boxId, u8 position)
 
 static void PurgeMonOrBoxMon(u8 boxId, u8 position)
 {
+    u16 item = ITEM_NONE;
+
     if (boxId == TOTAL_BOXES_COUNT)
+    {
+        if (OW_PC_RELEASE_ITEM >= GEN_8)
+            item = GetMonData(&gPlayerParty[position], MON_DATA_HELD_ITEM);
```

Comment:
Could you not avoid duplicating the code by writing it above the first if condition?

Reply:
Not really, because if the mon is in the party it's a different GetMonData function than when it's in the PC